libraries/chain/apply_context.cpp

```diff
@@ -184,10 +184,10 @@ void apply_context::exec_one()
       r.auth_sequence[auth.actor] = next_auth_sequence( auth.actor );
    }
 
-   trx_context.executed_action_receipt_digests.emplace_back( r.digest() );
-
    finalize_trace( trace, start );
 
+   trx_context.executed_action_receipt_digests.emplace_back( trace.digest_legacy() );
```

Comment:
This needs to be `digest_savanna()` for a proper IF block I believe. Please add a test that verifies this change.

Reply:
Do we not validate blocks during the transition that are not yet `proper IF`, but still use `block_state` and the new merkle tree?

Reply:
The validation is going to use this same code path which is using the legacy always.